src/main/java/com/example/productapi/controller/ProductController.java

```diff
@@ -37,9 +40,14 @@ public ResponseEntity<Product> getProductById(@PathVariable Long
                 .orElse(ResponseEntity.notFound().build());
     }
 
+    @GetMapping("/name")
+    public ResponseEntity<Product> findProductByName(@RequestParam String name) {
+        log.info("calling from findByName method");
```

Comment:
try to find via google, chatgpt better structure of logs (in terms of English and what need to write inside logs)

examples:
log.info("Invoking the findByName() method.");
log.info("Executing the findByName() method to retrieve data by name.");